types/jquery/JQuery.d.ts

```diff
@@ -7651,7 +7651,6 @@ $( "*", document.body ).click(function( event ) {
      * @param handler A function to execute when the event is triggered.
      * @see \`{@link https://api.jquery.com/on/ }\`
      * @since 1.7
-     * @deprecated ​ Deprecated. Use \`{@link JQuery.Event }\` in place of \`{@link JQueryEventObject }\`.
```

Comment:
It is not the "on" method but the use of JQueryEventObject that is deprecated, if you change your parameters/arguments to use jQuery.Event you won't receive a deprecation warning.

Reply:
Oh yeah!. I have seen why this function is referenced